x/ccv/provider/keeper/distribution.go

```diff
@@ -267,3 +280,23 @@ func (k Keeper) IdentifyConsumerChainIDFromIBCPacket(ctx sdk.Context, packet cha
 
 	return chainID, nil
 }
+
+// HandleSetConsumerCommissionRate sets a per-consumer chain commission rate for the given provider address
+// on the condition that the given consumer chain exists.
+func (k Keeper) HandleSetConsumerCommissionRate(ctx sdk.Context, chainID string, providerAddr types.ProviderConsAddress, commissionRate sdk.Dec) error {
+	// check that the consumer chain exists
+	if !k.IsConsumerProposedOrRegistered(ctx, chainID) {
```

Comment:
Doesn't seem necessary

Reply:
Removed.

Reply:
I don't think we should remove this. What's the benefit of setting a comission rate for chains that do not exist?
Also, leaving this check would be consistent with what `AssignConsumerKey` is doing.

Reply:
I agree that we should keep it. If there is no consumer chain, most likely the user made a mistake and they'll not be aware of that if the transaction passes.